src/__tests__/test-utils.ts

```diff
@@ -139,29 +138,39 @@ export class MockEventProducerService implements EventProducerService {
   destroy(): void {}
 }
 
+export async function generateRequest(override: Partial<Request>) {
```

Comment:
Optional suggestion. Feel free to disregard. It might be better to create `IRequest` interface.

Reply:
UPD: Maybe not, since Request is a pretty dumb class now.